cdc/cdc/capture/capture.go

```diff
@@ -434,6 +434,27 @@ func (c *Capture) AsyncClose() {
 		c.regionCache.Close()
 		c.regionCache = nil
 	}
+
+	c.cancel()
+
+	if c.etcdClient != nil {
+		timeoutCtx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		if err := c.election.Resign(timeoutCtx); err != nil {
```

Comment:
There is another `Resign` in `campaignOwner`. Why we call again here ?

Reply:
When exiting normally, `cancel` will be called, later `owner routine` & `processor routine` exit  normally. For a `non-owner` 's `owner routine`, it should return from `campagin`.

In etcd `campagin`, when `cancel` is called, `owner routine` will return from `waitDelete`[g](https://github.com/etcd-io/etcd/blob/main/client/v3/concurrency/election.go#L93) and call `resign`[g](https://github.com/etcd-io/etcd/blob/main/client/v3/concurrency/election.go#L98)`, but now ctx is `cancel`, so `resign` will not work.

So we should call it agait.

Reply:
From detail, https://github.com/pingcap/tiflow/pull/6284